src/SFA.DAS.AssessorService.Application.Api.Client/Clients/ApiClientBase.cs

```diff
@@ -86,41 +86,54 @@ private static HttpRequestException CreateRequestException(HttpRequestMessage fa
 
         protected async Task<T> RequestAndDeserialiseAsync<T>(HttpRequestMessage request, string message = null)
         {
-            HttpRequestMessage clonedRequest = null;
-
-            var result = await _retryPolicy.ExecuteAsync(async () =>
+            try
             {
-                clonedRequest = new HttpRequestMessage(request.Method, request.RequestUri);
-                clonedRequest.Headers.Add("Accept", "application/json");
-                clonedRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", TokenService.GetToken());
+                HttpRequestMessage clonedRequest = null;
 
-                return await HttpClient.SendAsync(clonedRequest);
+                var result = await _retryPolicy.ExecuteAsync(async () =>
+                {
+                    clonedRequest = new HttpRequestMessage(request.Method, request.RequestUri);
+                    clonedRequest.Headers.Add("Accept", "application/json");
+                    clonedRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", TokenService.GetToken());
 
-            });
+                    return await HttpClient.SendAsync(clonedRequest);
 
-            if (result.StatusCode == HttpStatusCode.OK)
-            {
-                // NOTE: Struct values are valid JSON. For example: 'True'
-                var json = await result.Content.ReadAsStringAsync();
-                return await Task.Factory.StartNew<T>(() => JsonConvert.DeserializeObject<T>(json, JsonSettings));
-            }
+                });
 
-            if (result.StatusCode == HttpStatusCode.NotFound)
-            {
-                if (message == null)
+                if (result.StatusCode == HttpStatusCode.OK)
                 {
-                    if(!request.RequestUri.IsAbsoluteUri)
-                        message = "Could not find " + request.RequestUri;
-                    else
-                        message = "Could not find " + request.RequestUri.PathAndQuery;
+                    // NOTE: Struct values are valid JSON. For example: 'True'
+                    var json = await result.Content.ReadAsStringAsync();
+                    return await Task.Factory.StartNew<T>(() => JsonConvert.DeserializeObject<T>(json, JsonSettings));
                 }
```

Comment:
Should update this to use Task.Run instead of Task.Factory.StartNew<T>

Reply:
Keeps it simpler so done it. I'm not sure why GitHub flagged it as new code consider it was there previosuly down below 👇

Reply:
Keeps it simpler so done it. I'm not sure why GitHub flagged it as new code consider it was there previously up above ☝️